src/components/App/App.test.tsx

```diff
@@ -92,4 +97,46 @@ describe('Gamkenbot App', () => {
       expect(driver.get.startButton()).toBeDisabled();
     });
   });
+
+  describe('Change icon based on status', () => {
+    beforeAll(() => {
+      jest.spyOn<React>(React, 'useState').mockImplementation((initialValue) => [initialValue, jest.fn()]);
```

Comment:
the generic here seems to be a must because TypeScript's inference doesn't work well out of the box here :(